spec/lightstep_spec.rb

```diff
@@ -111,6 +111,14 @@ def init_callback_tracer(callback)
     span.finish
   end
 
+  it 'should allow tag-setting of bool and number at start_span time' do
+    tracer = init_test_tracer
+    span = tracer.start_span('my_span', tags: {'number_key' => 1, 'bool_key' => true})
+    expect(span.tags['number_key']).to eq('1')
+    expect(span.tags['bool_key']).to eq('true')
```

Comment:
An expanded version of this might be to add values of a bunch of types and just assert that all span tag values `is_a?(String)`.

Additionally, the code calls `#to_s` on the objects, but the test doesn't show what happens if you assign a value that doesn't have `#to_s` on it. The current code would throw a `NoMethodError`, which maybe isn't what's intended.

Reply:
As a Ruby Newbie, is there a convenient thing I can call that doesn't have `#to_s` other than just making a new class?

Reply:
I updated my comment because Object provides `#to_s`; but it just prints out the class and object id. So even a new class would still have `#to_s`; the only kind of class that wouldn't would be one that _doesn't_ inherit from Object, or which has been stripped of those methods.

This comes down to what the contract is for tag values. If it can be stated that no matter the object given, the span's value for the tag will always be whatever the `#to_s` value of the object is, then it doesn't really matter what the string ends up being, and so that class name and object id is sufficient.

I'd throw in some tests for what happens when you give other things like an array, a hash, and some random object (defined as a new class), just to make the contract clear.